graylog2-web-interface/src/components/permissions/EntityShareModal.jsx

```diff
@@ -0,0 +1,69 @@
+// @flow strict
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { useStore } from 'stores/connect';
+import { Spinner } from 'components/common';
+import { EntityShareStore, EntityShareActions } from 'stores/permissions/EntityShareStore';
+import BootstrapModalConfirm from 'components/bootstrap/BootstrapModalConfirm';
+
+import EntityShareSettings from './EntityShareSettings';
+
+type Props = {
+  description: string,
+  entityId: string,
+  entityTitle: string,
+  entityType: string,
+  onClose: () => void,
+};
+
+const _generateGRN = (id, type) => `grn::::${type}:${id}`;
```

Comment:
@linuspahl: We might to move this somewhere else. What do you think?

Reply:
Sure, maybe in an own file, like `logic/permissions/GRN`?